### DISCUSSION TITLE
dynamic import stuck loading

### DISCUSSION THREAD

**Discussion by Pagnito** [vercel/next.js#33794] [Help]
Hey folks,
I am trying to load my components dynamically, so I can configure the layout in a config file, but my dynamic imports are stuck in loading

```
import React from 'react';
import {app} from '../app.config';
import dynamic from 'next/dynamic'

const importComponent = (path) => dynamic(() => import(path));

export default function Organizer(props) {
  let Layout = app.layout.landing.map(({type, style}) => {
    let path = `./${type}/${style}`
    let Component = importComponent(''+path);
    return <Component key={type+style}/>
  })
  return Layout
}

```

**Comment by realdocweird** [comment]
same for me, can't find a solution, did you found any?

**Comment by 2stash** [comment]
Just looking at the docs:

Note: In import('path/to/component'), the path must be explicitly written. It can't be a template string nor a variable. 

https://nextjs.org/docs/advanced-features/dynamic-import

**Reply by realdocweird** [reply]
in my case the problem was not there, the problem was to add `assetPrefix: "."` in the next config file to have everything working in the static export correctly
